tests/tests/integration_tests.rs

```diff
@@ -393,34 +393,40 @@ async fn ws_server_cancels_stream_after_reset_conn() {
 	let server = WsServerBuilder::default().build("127.0.0.1:0").await.unwrap();
 	let server_url = format!("ws://{}", server.local_addr().unwrap());
 
-	let (tx, mut rx) = mpsc::channel(1);
+	let (tx, rx) = mpsc::channel(1);
 	let mut module = RpcModule::new(tx);
 
 	module
-		.register_subscription("subscribe_never_produce", "n", "unsubscribe_never_produce", |_, sink, mut tx| {
-			// create stream that doesn't produce items.
-			let stream = futures::stream::empty::<usize>();
```

Comment:
this test was flaky I don't know what I was thinking.

Reply:
Further, this could complete for `notify_one` successfully if the `receiver` from the connection is closed once the a new item is produced on the stream.

so really tricky test but I added sleep for one hour so should be "okeyisch" really tricky to test this.